include/transport/time_queue.h

```diff
@@ -352,60 +312,74 @@ namespace qtransport {
          */
         std::optional<T> front()
         {
-            std::lock_guard<std::mutex> lock(_mutex);
-
             const tick_type ticks = advance();
 
             while (_queue_index < _queue.size()) {
-                auto& [bucket_index, value_index, expiry_tick] = _queue.at(_queue_index);
-                auto& bucket = _buckets.at(bucket_index);
+                const auto& [bucket_index, value_index, expiry_tick] = _queue.at(_queue_index);
+                const auto& bucket = _buckets.at(bucket_index);
 
                 if (value_index >= bucket.size() || ticks > expiry_tick) {
+                    /**
+                     * TODO: Below log is only added for debugging right now. This should be removed when stable or
+                     *    when we have a logger.
+                     */
+                    std::cerr << "===> front Object has expired"
+                              << " queue_index: " <<_queue_index << " queue_size: " << _queue.size()
+                              << " value_index: " << value_index
+                              << " bucket_index: " << bucket_index
+                              << " bucket_size: " << bucket.size()
+                              << " tick_delta: " << _timer_ctx.delta
+                              << " " << ticks << " > " << expiry_tick
```

Comment:
Probably don't need this now, since without the spike detection this might spam.

Reply:
Removed.